launchprocess.q

```diff
@@ -0,0 +1,45 @@
+/function which execute system commands from q, and is tailored to deal with the exit codes of lproc.sh:
+/cmd is a string and params are the process parameters which are passed to launch() 
+syscomm:{[params;cmd]
+
+    prevexitcode:first `$system cmd,"; echo $?";
+    $[`0~prevexitcode;
+        .lg.o[`launchprocess;"Successful execution: Starting ",params[`procname]];
+      `1~prevexitcode;
+        .lg.e[`launchprocess;"Error: ",params[`procname]," failed to start"];
+      `2~prevexitcode;
+        .lg.e[`launchprocess;"Error: ",params[`procname]," already running"];
+      .lg.e[`launchprocess;"Unknown error encountered, ",params[`procname]," failed to launch"]]   }
+
+
+
+/function which lets us call launchprocess.sh from inside a TorQ process
+/it takes a dictionary of parameters which will be passed to launchprocess.sh, i.e "-procname rdb1 -proctype rdb" etc.
+
+/ not sure how namespace declaration works     \d .process
+
+launch:{ [params]
+    /exit immediately if process name and type aren't provided
+    if[not 11b~`procname`proctype in key params;0N!
+        .lg.e[`launchprocess;"Process name and type must be provided"];
+        :()];
+    
+    /set default values with .Q.def and string the result for the startline:
+    /for now, let the set of parameters which need to be provided with defaults be:
+    /(procname,proctype),U,localtime
+
+    deflt:`procname`proctype`U`localtime!(`;`;`$getenv[`KDBAPPCONFIG],"/passwords/accesslist.txt";1);
+    params:string each .Q.def[deflt] params; 
+   
+    $[4<count params;
+          [ extras:(-1*(count params)-4)#params; 
```

Comment:
will this work if e.g. `procname` is passed as 5th arg in `params` dict? can't remember if `.Q.def` reorders things, if it does this is maybe fine

but it might be easier to just do the same for all the args, rather than excluding the default ones and then pulling them out manually below?

Reply:
.Q.def does reorder the dictionary in the way that I needed, fortunately.

I see what you're saying, not sure why I got caught up in handling them differently to be honest.